omg_local_planner/include/omg_local_planner/omg_planner_ros.h

```diff
@@ -0,0 +1,140 @@
+  ~OMGPlannerROS();
+
+  /**
+   * @brief Given the current position, orientation, and velocity of the robot,
+   * compute velocity commands to send to the base.
+   * @param cmd_vel The velocity command to be passed to the robot base.
+   * @return True if a valid trajectory was found, false otherwise.
+   */
+  bool computeVelocityCommands(geometry_msgs::Twist& cmd_vel);
+
+  /**
+   * @brief setPlan Set the plan that the planner is following.
+   * @param orig_global_plan The plan to pass to the planner.
+   * @return True if the plan was updated successfully, false otherwise.
+   */
+  bool setPlan(const std::vector<geometry_msgs::PoseStamped>& orig_global_plan);
+
+  /**
+   * @brief Check if the goal pose has been achieved.
+   * @return True if achieved, false otherwise.
+   */
+  bool isGoalReached();
+
+  /**
+   * @brief Check if the local planner is initialized.
+   * @return True if initialized, false otherwise.
+   */
+  bool isInitialized();
+
+ private:
+  /**
+   * @brief Publish the local plan to be followed.
+   * @param path A set of poses composing the plan.
+   */
+  void publishLocalPlan(std::vector<geometry_msgs::PoseStamped>& path);
+
+  /**
+   * @brief Publish the global plan to be followed.
+   * @param path A set of poses composing the plan.
+   */
+  void publishGlobalPlan(std::vector<geometry_msgs::PoseStamped>& path);
+
+  bool initialized_;  ///< @brief Holds the planner initialization status.
+
+  /**
+   * @brief odom_helper_ Class used to provide odometry information.
+   */
+  base_local_planner::OdometryHelperRos odom_helper_;
+
+  tf::TransformListener* tf_;  ///< @brief Used for transforming point clouds.
+
+  costmap_2d::Costmap2DROS* costmap_ros_;  ///< @brief The global costmap.
+  tf::Stamped<tf::Pose>
+      current_pose_;  ///< @brief The current pose of the robot.
+  ros::Publisher g_plan_pub_; ///< @brief Global plan publisher.
+  ros::Publisher l_plan_pub_; ///< @brief Local plan publisher.
+
+
+  /**
+   * @brief goal_reached_client_ Client for calling the goal_reached service.
+   */
+  ros::ServiceClient goal_reached_client_;
+
+  /**
+   * @brief goal_reached_client_ Client for calling the set_plan service.
+   */
+  ros::ServiceClient set_plan_client_;
+
+  /**
+   * @brief goal_reached_client_ Client for calling the initialize service.
+   */
+  ros::ServiceClient initialize_client_;
+
+  /**
+   * @brief goal_reached_client_ Client for calling the compute_velocity service.
+   */
+  ros::ServiceClient compute_velocity_client_;
+
+  /**
+   * @brief goal_reached_srv_ Service name.
+   */
+  const std::string goal_reached_srv_ = "goal_reached";
+
+  /**
+   * @brief goal_reached_srv_ Service name.
+   */
+  const std::string set_plan_srv_ = "set_plan";
+
+  /**
+   * @brief goal_reached_srv_ Service name.
+   */
+  const std::string initialize_srv_ = "initialize";
+
+  /**
+   * @brief goal_reached_srv_ Service name.
+   */
+  const std::string compute_velocity_srv_ = "compute_velocity";
```

Comment:
Nit: constness can be made more explicit by variable names. Either all caps or `kComputeVelocitySrv_` (Google style).